scipy/io/mmio.py

```diff
@@ -505,7 +532,7 @@ def _parse_body(self, stream):
         return a
 
     #---------------------------------------------------------------------------
-    def _write(self, stream, a, comment='', field=None, precision=None):
+    def _write(self, stream, a, comment='', field=None, precision=None, symmetry=None):
```

Comment:
Should checking for symmetry be the default?  Presumably this can be slow on large matrices?

I don't follow the logic of `rep == self.FORMAT_ARRY` below, but am I right in thinking that the previous default was to check symmetry for array-like input, but not for sparse input?